discovery-provider/src/queries/get_playlists.py

```diff
@@ -2,23 +2,42 @@
 import sqlalchemy
 from sqlalchemy import desc
 
+from flask.globals import request
 from src import exceptions
 from src.models import Playlist, RepostType, SaveType
 from src.utils import helpers
 from src.utils.db_session import get_db_read_replica
-from src.queries.query_helpers import get_current_user_id, paginate_query, \
+from src.queries.query_helpers import paginate_query, \
   populate_playlist_metadata, get_users_ids, get_users_by_id
+from src.utils.redis_cache import extract_key, use_redis_cache
 
 logger = logging.getLogger(__name__)
 
+UNPOPULATED_PLAYLIST_CACHE_DURATION_SEC = 10
+
+def make_cache_key(args):
+    cache_keys = {
+        "user_id": args.get("user_id"),
+        "with_users": args.get("with_users")
+    }
+
+    if args.get("playlist_id"):
+        ids = args.get("playlist_id")
+        ids = map(str, ids)
+        ids = ",".join(ids)
+        cache_keys["playlist_id"] = ids
+
+    key = extract_key(f"unpopulated-playlist:{request.path}", cache_keys.items())
+    return key
+
 def get_playlists(args):
     playlists = []
-    current_user_id = get_current_user_id(required=False)
-    filter_out_private_playlists = True
+    current_user_id = args.get("current_user_id")
```

Comment:
If i'm reading this right, if we push this change out, current clients will break because they aren't passing in the current_user_id correctly anymore right?

Reply:
as discussed on slack, should be good bc i'm pulling it out in the routing layer